File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/ModelInfoProviderFactory.java
package org.cqframework.cql.cql2elm;

import java.util.Iterator;
import java.util.ServiceLoader;
import org.hl7.cql.model.ModelInfoProvider;

public class ModelInfoProviderFactory {
    private ModelInfoProviderFactory() {}

    public static Iterator<ModelInfoProvider> providers(boolean refresh) {
        var loader = ServiceLoader.load(ModelInfoProvider.class);
        if (refresh) {
            loader.reload();
        }

        return loader.iterator();
    }
}

File: Src/java/cql-to-elm/src/main/kotlin/org/cqframework/cql/cql2elm/ModelInfoProviderFactory.kt
package org.cqframework.cql.cql2elm

import java.util.*
import org.hl7.cql.model.ModelInfoProvider

object ModelInfoProviderFactory {
    fun providers(refresh: Boolean): Iterator<ModelInfoProvider> {
        val loader = ServiceLoader.load(ModelInfoProvider::class.java)
        if (refresh) {
            loader.reload()
        }
        return loader.iterator()
    }
}
